pykusto/client.py

```diff
@@ -48,50 +58,94 @@ def get_cluster_name(self) -> str:
     def _get_client_for_cluster(cluster: str) -> KustoClient:
         return KustoClient(KustoConnectionStringBuilder.with_aad_device_authentication(cluster))
 
+    def _internal_get_items(self) -> Dict[str, 'Database']:
```

Comment:
I don't like that each layer brings all of it's sublayers recursively.. There is a code duplication and some possible concurrency issues.. For example, it is unclear from the code whether a Database is fetching only it's tables or tables belonging to other db's..
Why not bringing only the databases in the client object, only the tables in the database object and only the columns in the table object?

Reply:
That would involve multiple queries instead of just one. In our own use case it would be hundreds of queries, and doesn't seem reasonable.
Where is there code duplication here? What are the concurrency issues?
Each level retrieves everything under that level (a cluster retrieves all its databases tables and columns, a database all its tables and columns, a table all its columns). Why would anyone assume a database retrieves tables beloning to other dbs? that doesn't make sense.
I added some documentation to make it clearer.

Reply:
Added comments clarifying the queries in "_internal_get_items" implementations